### ISSUE TITLE
Ruby 3.1 keyword omission leads to conflicting default guidance

### ISSUE THREAD

**Issue by dduugg** [rubocop/rubocop#10292]
## Expected behavior

The hash/keyword value omission [feature](https://bugs.ruby-lang.org/issues/14579) in Ruby 3.1 allows one to invoke a method with `bar(test:)` as shorthand for `bar(test: test)`. Here's an example in context:

```ruby
# frozen_string_literal: true

# test class
class A
  def self.foo(test)
    bar(test:)
  end

  def self.bar(test:)
    puts test
  end
end

puts A.foo('test') # => test
```

I'd expect some version of this example to pass under a default rubocop config.

## Actual behavior

The above example triggers always a rubocop violation, either `Layout/SpaceAfterColon` or `Layout/SpaceInsideParens` (or somehow both, when using autocorrect):

```
$ rubocop a.rb -A
Inspecting 1 file
C

Offenses:

a.rb:6:13: C: [Corrected] Layout/SpaceAfterColon: Space missing after colon.
    bar(test:)
            ^
a.rb:6:14: C: [Corrected] Layout/SpaceInsideParens: Space inside parentheses detected.
    bar(test: )
             ^

0 files inspected, 2 offenses detected, 2 offenses corrected
```

## Steps to reproduce the problem

☝️ 

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ rubocop -V
1.23.0 (using Parser 3.0.3.2, rubocop-ast 1.14.0, running on ruby 3.1.0 x86_64-darwin21)
  - rubocop-performance 1.12.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.6.0
```

**Comment by koic** [comment]
I'm working on that feature now. Partially solved by https://github.com/rubocop/rubocop/pull/10291 and I will see overall :-)

**Comment by dduugg** [comment]
Nice! (Sorry if this was a spammy ticket, I searched issues but not PRs 🙈 )

**Comment by koic** [comment]
No problem. Thank you for your early feedback!

**Comment by koic** [comment]
I merged #10291 to resolve this issue.
